### ISSUE TITLE
Chrome is logging [Violation] errors in verbose console mode

### ISSUE THREAD

**Issue by etpinard** [plotly/plotly.js#1795]
First reported https://community.plot.ly/t/violation-at-console/4518/7 

To see it, open up https://codepen.io/etpinard/pen/ZyLQYy and turn on _Verbose_ mode

![image](https://user-images.githubusercontent.com/6675409/27242962-a30adf2a-52ad-11e7-8b7d-38ed58cc8a9e.png)

The error message points to this [line](https://github.com/plotly/plotly.js/blob/2940d950d7e39c1be44462681ddd1c2139a0a649/src/plots/cartesian/dragbox.js#L440) in `dragbox.js`.

```
[Violation] Added non-passive event listener to a scroll-blocking 'wheel' event. 
Consider marking event handler as 'passive' to make the page more responsive.
```

Observed in Chrome 59 and 58.

---

Anyone know what this is about?

**Comment by rreusser** [comment]
I got the sense maybe they just added this to escalate the awareness of this general issue. I think it has to do with whether scrolling bubbles up to the window _and then_ executes the callback or whether it executes the callback _and then_ bubbles up to the window. Presumably it's more expensive UI-responsiveness-wise to bubble up to the window first rather than blocking scroll. Or something. See: https://developers.google.com/web/updates/2016/06/passive-event-listeners

**Comment by etpinard** [comment]
Related to https://github.com/plotly/plotly.js/issues/1240

**Comment by rreusser** [comment]
Let's see… this is a pretty good description of how to detect and apply it: https://github.com/WICG/EventListenerOptions/blob/gh-pages/explainer.md

**Comment by etpinard** [comment]
cc @dfcreative , which might know what this is about.

**Comment by efaden** [comment]
Did this ever get fixed...?  My events are crashing when I try to perform some logic... I just get TONS of events.

**Comment by etpinard** [comment]
@dfcreative you're going for this?

**Comment by brian428** [comment]
Any thoughts on getting this fixed? We're showing multiple charts on the screen at once and these non-passive listener violations are flooding the console.

![image](https://user-images.githubusercontent.com/140338/33910745-d39a1092-df5d-11e7-8f90-6a13d152dd8e.png)

**Comment by etpinard** [comment]
Unfortunately, this ticket fell off our short-term priority list. We'll gladly accept PRs from whoever is comfortable for the [specs](https://github.com/WICG/EventListenerOptions/blob/gh-pages/explainer.md) mentioned above.

In the meantime, the only workaround I can propose is to turn off Chrome's verbose logging level.

**Comment by etpinard** [comment]
Interesting report in https://community.plot.ly/t/scrollzoom-doesnt-work-in-chrome/7356 that might be related to this ticket.

**Comment by dfcreative** [comment]
I am not sure we can fix that. Both cases where we attach event listener (`scrollZoom` and touch panning), we need it blocking screen scrolling, therefore we cannot make it passive.

**Comment by etpinard** [comment]
Ok. We can't _just_ close this as clearly some people are finding this annoying.

I wonder if there's a way to turn off the those console warning without altering the functionality.

**Comment by dfcreative** [comment]
#2251 fixes that, although exposes event listener on DOM node.

**Comment by oscar-broman** [comment]
There is also a dependency on [mouse-wheel](https://github.com/mikolalysenko/mouse-wheel), which has not been patched.

It is used in `src/plots/gl3d/camera.js`, for example.

**Comment by jacobq** [comment]
Was this closed because it was fixed or because it's being tracked in #1240?
I am still seeing violation warnings in the console with plotly.js `v1.35.2` that look like this:

> plotly.js:32018 [Violation] Added non-passive event listener to a scroll-blocking 'touchstart' event. Consider marking event handler as 'passive' to make the page more responsive. See https://www.chromestatus.com/feature/5745543795965952

**Comment by etpinard** [comment]
@jacobq we attempted a fix in #2251, so that's why thing think is closed.

Can you please share a reproducible example to help us debug?

**Comment by jacobq** [comment]
I am seeing it in an electron v2.0.0-beta.1 app, which uses Chrome 61. (Happens when I `mousedown` on a plot.) However, I haven't been able to reproduce it in a browser yet. It's possible that it's actually being [caused by an extension](https://stackoverflow.com/questions/46717866/violation-added-non-passive-event-listener-when-no-javascript-present) or something. I can't spend more time playing with it today, but if I am able to come up with a simple jsbin demo for you later then I'll be sure to share it here.

**Comment by oscar-broman** [comment]
My reply clearly states the root of the problem:
https://github.com/mikolalysenko/mouse-wheel

Look at the bottom of wheel.js in that repo.

**Comment by jacobq** [comment]
@oscar-broman I don't think that would explain the message I encountered though, right? In my case it was regarding `touchstart` rather than `wheel` events.

**Comment by etpinard** [comment]
Right and `mouse-wheel` is only used in gl3d and pre-regl gl2d graphs. Perhaps @jacobq discovered another problematic case.

**Comment by jacobq** [comment]
OK, I broke my own promise and spent more time digging into this one.
Minimal demo here:
https://github.com/jacobq/electron-plotly-hello-world
Since the exact same app doesn't seem to generate these warnings when opened in Chrome 64 (see [`non-electron` branch](https://github.com/jacobq/electron-plotly-hello-world/tree/non-electron)),
I think this is something related to Electron (it could still be between Chrome 61 and 64 -- I don't have an older Chrome instance handy to check -- but I doubt that's it).

**Comment by etpinard** [comment]
Ok thanks! I guess the least I can do at this stage is to reopen this ticket.

**Comment by jacobq** [comment]
:+1: Also feel free to clone and run my demo to see it for yourself. Maybe it will be more obvious to you.
I don't know if this is truly related to it but I'm adding a link here just in case: https://github.com/electron/electron/issues/8725

**Comment by alexandertu** [comment]
Any updates on this?

**All** examples of choropleth from https://plot.ly/javascript/choropleth-maps generates multiple messages: 

> [Violation] Added non-passive event listener to a scroll-blocking 'touchstart' event. Consider marking event handler as 'passive' to make the page more responsive. See https://www.chromestatus.com/feature/5745543795965952

My version of Chrome is 66. The messages are generated when I double click on any country.
However, the example with Mapbox doesn't generate any messages...

**Comment by boyswan** [comment]
Also wondering the status of this!

**Comment by jonborchardt** [comment]
Also wondering the status of this!

**Comment by bikramtpa** [comment]
also wondering

**Comment by dy** [comment]
Blocked by https://github.com/mikolalysenko/mouse-wheel/pull/7

**Comment by jackparmer** [comment]
This issue has been tagged with `NEEDS SPON$OR`

A [community PR](https://github.com/plotly/plotly.js/pulls?q=is%3Apr+is%3Aopen+label%3A%22type%3A+community%22) for this feature would certainly be welcome, but our experience is deeper features like this are difficult to complete without the Plotly maintainers leading the effort.

**Sponsorship range: $5k-$10k**

What Sponsorship includes:
- Completion of this feature to the Sponsor's satisfaction, in a manner coherent with the rest of the Plotly.js library and API
- Tests for this feature
- Long-term support (continued support of this feature in the latest version of Plotly.js)
- Documentation at [plotly.com/javascript](https://plotly.com/javascript/)
- Possibility of integrating this feature with Plotly Graphing Libraries (Python, R, F#, Julia, MATLAB, etc)
- Possibility of integrating this feature with Dash
- Feature announcement on [community.plotly.com](https://community.plotly.com/tag/announcements) with shout out to Sponsor (or can remain anonymous)
- Gratification of advancing the world's most downloaded, interactive scientific graphing libraries (>50M downloads across supported languages)

Please include the link to this issue when [contacting us](https://plotly.com/get-pricing/) to discuss.

**Comment by pavel-vorobev-qs** [comment]
Consider to use other lib because every bug require sponsor and docks are piece of SPONSOR$HIP.

**Comment by gvwilson** [comment]
Hi - this issue has been sitting for a while, so as part of our effort to tidy up our public repositories I'm going to close it. If it's still a concern, we'd be grateful if you could open a new issue (with a short reproducible example if appropriate) so that we can add it to our stack. Cheers - @gvwilson
